tests/fips/stunnel.pm

```diff
@@ -0,0 +1,101 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: stunnel tests in FIPS mode
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#108608, tc#1769815
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use utils;
+use mm_tests;
+use mmapi 'wait_for_children';
+
+my $hostname = get_var('HOSTNAME');
+# Set vnc password
+my $password = '123456';
+
+sub conf_stunnel_vnc {
+    my $stunnel_config = <<EOF;
+client = no
+chroot = /var/lib/stunnel/
+pid = /var/run/stunnel.pid
+socket = l:TCP_NODELAY=1
+socket = r:TCP_NODELAY=1
+cert = /etc/stunnel/stunnel.pem
+
+fips =yes
+
+[VNC]
+accept = 15905
+connect = 5905
+EOF
+    assert_script_run("echo '$stunnel_config' >>  /etc/stunnel/stunnel.conf");
+    if ($hostname =~ /client|slave/) {
+        assert_script_run q(sed -i 's/^client = no/client = yes/' /etc/stunnel/stunnel.conf);
+        assert_script_run q(sed -i 's/^connect = 5905/connect = 10.0.2.101:15905/' /etc/stunnel/stunnel.conf);
```

Comment:
Nice comments,

Let me explain why I don't set variable for the ports and ip addr here:

1. for the port, we try to modify the file use shell script, then I will use one more variable for this, it can't give clear info there, at the same time, I only use this port twice and <3, it is safe to use hard code
2. for ip addr, thanks for the comments, mm tests use this hard code ip for server and 10.0.2.102 for client, it is vlan ip addr and will not affect by other jobs.